csharp/src/Ice/TcpConnector.cs

```diff
@@ -9,75 +9,65 @@ namespace ZeroC.Ice
     internal sealed class TcpConnector : IConnector
```

Comment:
We should probably convert IConnector into an abstract class (just like Endpoint is an abstract class) since:
 - we don't need multiple inheritance (in all likelihood)
 - the abstract class can provide a static operator ==

Could be postponed until a larger transport refactoring.

Reply:
OK for doing this in another PR. Just wondering though... why provide operator== if we don't use it? Also, should the connector base class compare the endpoint Transport? I noticed the Endpoint base class doesn't compare the Transport, we only compare it in IPEndpoint so I guess to be consistent we should compare the Transport in TcpConnector?

Reply:
> why provide operator== if we don't use it?
Currently we can't provide operator == because the base connector is an interface. Once it's an abstract class, we can provide it and use it. 

> I noticed the Endpoint base class doesn't compare the Transport
I think that's a bug in Endpoint: it should compare and hash Transport.